README.md

```diff
@@ -1,2 +1,114 @@
+
+## Getting started
+
+This section covers how to install Istio mTLS and gateway private keys protection with SGX
+### Create signer
+
+```sh
+export CA_SIGNER_NAME=sgx-signer
+cat << EOF | kubectl create -f -
+apiVersion: tcs.intel.com/v1alpha1
+kind: TCSClusterIssuer
+metadata:
+    name: $CA_SIGNER_NAME
+spec:
+    secretName: ${CA_SIGNER_NAME}-secret
+    # If using quoteattestaion, set selfSign as false
+    # selfSign: false
+EOF
+```
+
+```sh
+# Get CA Cert and replace it in ./deployment/istio-configs/istio-hsm-config.yaml
+kubectl get secret -n tcs-issuer ${CA_SIGNER_NAME}-secret -o jsonpath='{.data.tls\.crt}' |base64 -d | sed -e 's;\(.*\);        \1;g'
+```
+### Protect the private keys of workloads with HSM
+1. Install Istio
+
+```sh
+istioctl install -f ./deployment/istio-configs/istio-hsm-config.yaml -y
+```
+
+> NOTE: You can also customize the `istio-hsm-config.yaml` according to your needs. If you want do the quote verification, you should set the `NEED_QUOTE` env as `true`. And if you are using the TCS v1alpha1 api, you can set the `RANDOM_NONCE` as `false`.
+
+
+
+2. Verifiy the Istio is ready
+
+By deault, `Istio` will be installed in the `istio-system` namespce
+
+```sh
+# Ensure that the pod is running state
+$ kubectl get po -n istio-system
+NAME                                    READY   STATUS    RESTARTS   AGE
+istio-ingressgateway-6cd77bf4bf-t4cwj   1/1     Running   0          70m
+istiod-6cf88b78dc-dthpw                 1/1     Running   0          70m
+```
+
+3. Create sleep and httpbin deployment:
+> NOTE: If you want use the sds-custom injection template, you need to set the annotations `inject.istio.io/templates` for both `sidecar` and `sgx`. And the ClusterRole is also required.
+```sh
+kubectl apply -f <(istioctl kube-inject -f ./deployment/istio-configs/sleep-hsm.yaml )
+kubectl apply -f <(istioctl kube-inject -f ./deployment/istio-configs/httpbin-hsm.yaml )
+```
+
+4. Successful deployment looks like this:
+
+```sh
+$ kubectl get po
+NAME                       READY   STATUS    RESTARTS   AGE
+httpbin-5f6bf4d4d9-5jxj8   3/3     Running   0          30s
+sleep-57bc8d74fc-2lw4n     3/3     Running   0          7s
+```
+5. Test pod resources:
+
+```sh
+$ kubectl exec "$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})" -c sleep -- curl -v -s http://httpbin.default:8000/headers | grep X-Forwarded-Client-Cert
+    "X-Forwarded-Client-Cert": "By=spiffe://cluster.local/ns/default/sa/httpbin;Hash=2875ce095572f8a12b6080213f7789bfb699099b83e8ea2889a2d7b3eb9523e6;Subject=\"CN=SGX based workload,O=Intel(R) Corporation\";URI=spiffe://cluster.local/ns/default/sa/sleep"
+
+```
+
+The above `httpbin` and `sleep` applications have enabled SGX and store the private keys inside SGX enclave, completed the TLS handshake and established a connection with each other.
```

Comment:
add a step to dump envoy config to demonstrate the private key has been switched to our private-key-provider